include/url.h

```diff
@@ -79,6 +79,11 @@ namespace Url
          */
         Url& abspath();
 
+        /**
+         * Evaluate 'other' relative to this URL, placing the result in this object.
+         */
+        Url& relative(const std::string& other);
```

Comment:
I'm tempted to not make this a mutating function since you may want to use the same base multiple times.